### PR TITLE
DHFPROD-2464: update version compatibility for DH

### PR DESCRIPTION
**Pull request by aebadirad**

### CODE REVIEW

**Comment by aebadirad** [comment]
This is only for 5.0.0, 5.0.1 will bring compatibility back. This is just to block 5.0.0 from installing onto a known incompatibility for an unreleased version of MarkLogic.

Though in the future, yes, we should use test assertions

**Comment by srinathgit** [comment]
I think we can remove the class CMASettings since 5.0 is supported in versions > 9.0-7 but that can be done later as well. I can create a task for it post 5.0

**Comment by aebadirad** [comment]
Yes, let's address that (and the upgrade to ml-gradle 3.14) post 5.0

**Comment by bsrikan** [comment]
There are failing tests. I hope this doesnt get merged till those are fixed.

**Comment by aebadirad** [comment]
@bsrikan these are the automated UI timeout failures. Can't do much about that, core code tests passed though.

**Comment by bsrikan** [comment]
https://jenkins.marklogic.com/blue/organizations/jenkins/Datahub_CI/detail/PR-2448/2/tests
I see 10 failed tests in core DataHubTest

**Comment by aebadirad** [comment]
> 
> 
> https://jenkins.marklogic.com/blue/organizations/jenkins/Datahub_CI/detail/PR-2448/2/tests
> I see 10 failed tests in core DataHubTest

Ah, yeah. Fixing that. Whoops. My bad.

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java [hunk 1]
```diff
@@ -273,13 +274,14 @@ public boolean isServerVersionValid(String versionString) {
...
+                //Future proofing against 9.10 since we have an ES incompatibility that needs resolving
+                if (!isNightly && (ver < 9070 || ver > 9095)) {
                     return false;
                 }
             }
             if (isNightly) {
                 String dateString = versionString.replaceAll("[^-]+-(\\d{4})(\\d{2})(\\d{2})", "$1-$2-$3");
-                Date minDate = new GregorianCalendar(2018, 4, 5).getTime();
+                Date minDate = new GregorianCalendar(2018, 11, 5).getTime();
                 Date date = new SimpleDateFormat("y-M-d").parse(dateString);
```
Comment (srinathgit) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java]: This corresponds to 12/5/2018 's nightly(GregorianCalendar's month is from 0 to 11).  So this test has to be changed
public void testValidateServer9nightly() {
        assertTrue(dh.isServerVersionValid("9.0-20181110"));
}

Comment (akshaysonvane) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java]: Resolved via https://github.com/marklogic/marklogic-data-hub/pull/2448/commits/9ae168e1eb0a3fde726f6dd18fcf8ef62d89d9c0

**Pull request merged**